propolis/src/instance.rs

```diff
@@ -9,14 +11,27 @@ use crate::inventory::Inventory;
 use crate::vcpu::VcpuRunFunc;
 use crate::vmm::*;
 
+/// States of operation for an instance.
 #[derive(Copy, Clone, Eq, PartialEq, Debug)]
 pub enum State {
+    /// Initial state. Instances cannot return to this state
+    /// after transitioning away from it.
     Initialize,
+    /// The instance is booting. Instances automatically
+    /// transition to boot after initialization, unless they're
+    /// being explicitly destroyed.
```

Comment:
For the time being, I would avoid the commentary about the automatic state changing, considering this is still pretty early and subject to change